linker-interface/shared/src/main/scala/org/scalajs/linker/interface/Report.scala

```diff
@@ -0,0 +1,155 @@
+/*
+ * Scala.js (https://www.scala-js.org/)
+ *
+ * Copyright EPFL.
+ *
+ * Licensed under Apache License 2.0
+ * (https://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package org.scalajs.linker.interface
+
+import java.io._
+
+import org.scalajs.ir.ScalaJSVersions
+
+import org.scalajs.linker.interface.unstable.ReportImpl
+
+/** Information about a linker run. */
+abstract class Report private[interface] {
+  /** Public modules resulting from linking.
+   *
+   *  An incremental linker must return all public modules here, even if some
+   *  were not re-written during this linking run.
+   */
+  def publicModules: Iterable[Report.Module]
```

Comment:
We should have an known-to-be-immutable data type, here. I think it should be `immutable.Seq[Report.Module]`, if not `List[Report.Module]`.

Reply:
I'd rather not commit to the ordering of the modules. Is `immutable.Iterable` OK?

Reply:
Yes, `immutable.Iterable` works too.